webapp/src/controllers/datasource.ts

```diff
@@ -493,7 +493,10 @@ export async function addDatasourceApi(req, res, next) {
 			: null
 	});
 
-	return dynamicResponse(req, res, 302, { redirect: `/${req.params.resourceSlug}/datasources` });
+	return dynamicResponse(req, res, 302, {
+		redirect: `/${req.params.resourceSlug}/datasources`,
+		toolId: addedTool.insertedId
```

Comment:
make sure you update your api docs and spec for this :wink:

you can push that to straight to develop